client/dfclient/src/gameplay_state.cpp

```diff
@@ -92,6 +95,40 @@ void GameplayState::LoadLevel() {
 		spritemap.insert(std::pair<uint16_t, std::string>(fb_Ti->gid(), fb_Ti->name()->str()));
 	}
 
+	// initialize tile layer
+	if (level->tilelayer()) {
+		auto width = level->tilelayer()->width();
+		auto height = level->tilelayer()->height();
+		std::vector<std::string> rows;
+		
+		std::stringstream ss(level->tilelayer()->data()->str());
+		std::string row;
+		while (std::getline(ss, row, '\n')) {
+			if(row.empty()) continue; // first row is empty
+			rows.push_back(row);
+		}
+
+		for (int i=0; i<height; ++i) {
+			std::vector<std::string> ids;
+
+			std::stringstream ss2(rows[i]);
+			std::string id;
+			while (std::getline(ss2, id, ',')) {
+				ids.push_back(id);
+			}
+
+			for (int j=0; j<width; ++j) {
+				auto spritename = spritemap[std::stoi(ids[j])];
+				if (spritename.empty()) continue;
+				auto tileSprite = std::make_unique<ncine::Sprite>(this->cameraNode.get(), _resources.textures[spritename].get(),
+					j * TILE_WIDTH, -i * TILE_HEIGHT);
+				tileSprite->setAnchorPoint(0, 1);
+				tileSprite->setLayer(TILE_LAYER);
+				this->nodes.push_back(std::move(tileSprite));
+			}
+		}
+	}
+
```

Comment:
Just my two cents: I feel we should avoid any custom parsing inside C++. Instead, we should parse the tilemap inside python and serialize it as a FlatBuffer-encoded array.

There is a dedicated library for parsing CSV included in python's stdlib which should make it easy: https://docs.python.org/3/library/csv.html